File: loops/notes.py
#Robert Nelson, Loops Notes Python

#1.What is a loop? 
    #A section of code that repeats multiple times
#2.What are the two types of loops?
    #for loop
nums = [1,1,2,3,5,8,13,21,34]

for num in nums:
    print(num)
    #while loop

x = 0

while x < 10:
    print(x)
    x+= 1
#3.What is iteration
    #That specific instance of the loop
    #iteration the amount of times you are looping through
#4.What are lists?
    # A list is just a variable that holds multiple values?
nums = [1,2,3,4,5,6,7]
siblings = ["Alex", "Katie", "Andrew", "Vienna", "Tia", "Treyson", "Xavier", "Hailey"]
print(nums) #prints while list, which is ugly
print(siblings[1]) #prints one item from the list

siblings[0] = "Jeremy" #renames user
print(siblings)

siblings.pop(0) #removes user
siblings.insert(0, "John")
print(siblings)
siblings.insert(1, "Gary")
#siblings.insert(2, "Joe", "Noah", "Zee")
#5.How do you make lists in python?
    #You use brackets, add in items with correct data types, and you have to have a comma between each item.
#6.How do you make for loops in python? 
for sibling in siblings:
    print(siblings)
for x in range(1, 5): #range stops before the max number which is 5
    print(x)
#7.How do you make while loops in python?
import random
x = 1 #variable to keep count of iteration
goose = random.randint(1,20)


while x <= 20:
    if x == goose:
        print("Goose!")
        break # tells loop to stop
    else:
        print("Duck")
    #print(x)
    x+= 1

#continue moves on to the next iteration without finishing
#8.How do you make lists in C?
#9.How do you make for loops in C?
#10.How do you make while loops in C?

File: loops/notes.c
//Robert Nelson, Loops Notes - C
#include <stdio.h>

int main(void){

    //1. What is a loop?
        //a section of code that repeats
    //2. What are the two types of loops?
        //while loops
        
    int start = 0;
    while(start < 5){
        printf("Hello\n");
        start++;
    }
    //For Loops
    int q;
    for(q=0;q<5;q++){
        printf("%d\n", q);
    }
    
    //3. What is iteration
        //repeating something with minor changes
    //4. What are arrays(lists)? 
        //an array (list) is a variable holding multiple values
    //8. How do you make arrays(lists) in C?
        //array items must be the same data type
    int grades[] = {88, 97, 100, 70, 72, 99, 61};
        // 1. set data type
        // 2. After naming put brackets and write the length of the list
        // 3. List is surronded by curly brackets {}
        // 4. Commas , between each item
    // print a single item from a list
    printf("CSP Grade: %d\n", grades[2]);

    // Change an item in the array
    grades[2] = 95;
    printf("CSP Grade: %d\n", grades[2]);
    //Size of array in bytes
    int size = sizeof(grades);
    // length in array items
    int length = sizeof(grades)/sizeof(grades[0]);
    printf("The array is %d\n", length);
    // array with strings
    //First brackets sets length of array(list)
    //Second brackets sets length of each string
    char movies[][20] = {"Cars", "Treasure Planet", "An American Tale", "Marley and Me", "The Avengers"};
    printf("The movie is %s!\n", movies[1]);
    int mlength = sizeof(movies)/sizeof(movies[0]);
    //9. How do you make for loops in C?
    // set the iterator, keeps track of time through the loop (best practice to set it as x or i)
    int x;
    // in parentheses (starting point; ending point; count by)
    for(x=0;x<=10;x++){
        printf("%d\n", x);
    }
    int m;
    for(m=0;m<mlength;m++){
        printf("%s\n", movies[m]);
    }
    //10. How do you make while loops in C?
    int w = 100;//start pount

    while(w>=0){//stop point
        printf("%d\n", w);
        w -= 10;//what we count by
    }
    return 0;
}